lib/test_lib.py

```diff
@@ -571,7 +571,7 @@ def SetupClients(self, nr_clients):
         fd.Flush()
 
         index.AddClient(fd)
-
+    index.Close()
```

Comment:
I found this because I switched over to using the client index to find clients.  self.SetupClients(x) creates and adds the clients to the index, but it doesn't actually get written back until index.Close() since this isn't using "with" to create the index.  From what I saw other unit tests were either building a separate client index or otherwise updating the client index separately.

The problem was that in the unit test I would use SetupClients and then run the flow, but the index would be empty so it couldn't find any clients to delete and the test would fail.  After adding index.Close() the index would get updated and the flow would successfully find the clients and delete them.